src/common/data/table-data/table-data-client/table-data-response.ts

```diff
@@ -1,4 +1,4 @@
-export interface TableDataResponse<T> {
+export interface TableDataResponse<T, U = unknown> {
   rows: Array<T>;
-  pageData?: Record<string, unknown>;
+  pageData?: Record<string, U>;
```

Comment:
this isn't quite right, pageData should be type U with default of Record<string, unknown>